### PR TITLE
Implement serialization for LinearRing.

### PR DESCRIPTION
**Pull request by dblock**
### Description

A user that enables trace-level logging with OpenSearch will run into a serialization error on geo tasks that they otherwise would not run into. It comes from [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/geo/GeoJson.java#L129). This also happens in a test, and https://github.com/opensearch-project/OpenSearch/pull/12783 attempts to ignore the failed serialization error. This PR implements JSON serialization for `LinearRing`.

Reproduce with

```
./gradlew ':server:test' --tests "org.opensearch.search.geo.GeoPointShapeQueryTests.testQueryLinearRing" -Dtests.opensearch.logger.level=TRACE
``` 

### Related Issues

Closes https://github.com/opensearch-project/OpenSearch/issues/11688.

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
LMK if you think it's better @shwetathareja @reta.

File: server/src/main/java/org/opensearch/action/search/SearchTask.java [hunk 1]
```diff
@@ -80,7 +80,11 @@ public SearchTask(
 
     @Override
     public final String getDescription() {
-        return descriptionSupplier.get();
+        try {
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/search/SearchTask.java]: @dblock so I've check the codebase and it seems like we have at least 4 consolidated places where `UnsupportedOperationException` is thrown all over the place:
 - GeoJson (JSON serialization)
 - GeometryIO (native stream serialization)
 - GeoShapeIndexer (indexer)
 - LegacyGeoShapeQueryProcessor / VectorGeoPointShapeQueryProcessor / LuceneGeometryCollector

 It seems like `LinearRing` is special case and should not be used as a top level geometry primitive, looking a bit more on how we could fix that.

**Comment by andrross** [comment]
I'll start with the caveat that I know nothing about GeoJson, but does this serialization conform with the [GeoJSON specification](https://datatracker.ietf.org/doc/html/rfc7946)? I spent a few minutes looking at this and I couldn't quite figure it out (again, I know nothing about this domain), but which of the [seven allowed geometries](https://datatracker.ietf.org/doc/html/rfc7946#section-1.4) does a linear ring serialize itself as since "linear ring" isn't one of the allowed geometries?

**Comment by dblock** [comment]
Maybe @nknize can take a look at this? I approached this problem from a very practical point of view: task description wants to serialize the linear ring so that the user can see what's going on in the trace, and serialization was missing. I don't know if there are negative consequences of introducing this implementation.

**Comment by dblock** [comment]
> ❌ Gradle check result for [2271a3a](https://github.com/opensearch-project/OpenSearch/commit/2271a3a02d127050d3573b75a75d4190efaa00cb): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/35599/)
> 
> Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green. Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

org.opensearch.indices.IndicesRequestCacheIT.testCacheWithInvalidation {p0={"opensearch.experimental.feature.pluggable.caching.enabled":"false"}} https://github.com/opensearch-project/OpenSearch/issues/12308

**Comment by andrross** [comment]
> I approached this problem from a very practical point of view:

This approach is completely fine for the "convert this structure to text to write into a log message" use case. The problem is the serialization is also used to create what I assume must be conforming GeoJSON potentially for communicating with other systems that expect conforming GeoJSON. Unfortunately I don't see any easy way to tease apart these use cases because the logging happens way up in the call stack while operating on very generic objects.

**Comment by dblock** [comment]
@andrross @reta LMK if you prefer https://github.com/opensearch-project/OpenSearch/pull/12834

**Comment by kkmr** [comment]
Looks like @rishabhmaurya  also posted a solution to the issue:  https://github.com/opensearch-project/OpenSearch/pull/12844. :-)

**Comment by dblock** [comment]
I'd like to leave this open for a bit, hoping someone (@nknize?) will have a strong opinion of why we should not have this serialization implementation for LinearRing.

**Comment by reta** [comment]
> @shwetathareja in [#12834 (comment)](https://github.com/opensearch-project/OpenSearch/pull/12834#discussion_r1535543560) is saying this is dead code and we should delete it.

Definitely let's delete it if it is dead (but I see it being used), may be what exactly is "dead" would good to clarify

**Comment by dblock** [comment]
Closing in favor of https://github.com/opensearch-project/OpenSearch/issues/12910.

**Pull request closed without merging**